tests/console/repo_orphaned_packages_check.pm

```diff
@@ -0,0 +1,37 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check for packages orphaned from present repos
+# Maintainer: Michal Nowak <mnowak@suse.com>
+
+use base 'consoletest';
+use strict;
+use testapi;
+use utils 'zypper_call';
+
+sub run {
+    select_console 'root-console';
+
+    # Pipe-separated list of packages we whitelist for orphan checking
+    my $orphan_whitelist = get_var('ORPHAN_WHITELIST', 'gpg-pubkey');
+    zypper_call('ref');
+    assert_script_run('
+ret=0
+for name in $(rpmquery -a --qf "%{NAME}\n" | grep -v -E ^"' . $orphan_whitelist . '"$)
+do
+  if ! zypper --no-refresh info $name | grep -wq ^Repository; then
+    echo "ERROR: $name package is not present in available repos"
+    ret=1
+  fi
+done
+if [ $ret -ne 0 ]; then false; fi', 600);
```

Comment:
Do you think it will be stable considering problems with typing? Just in case it's not, we could use [save_tmp_file](http://open.qa/api/testapi/#_save_tmp_file) to upload script and then execute in SUT after downloading it from the worker.

Reply:
Great idea. Tested on QEMU as well as Xen & Hyper-V.